keps/sig-cli/kustomize-secret-generator-plugins.md

```diff
@@ -0,0 +1,136 @@
+---
+title: Kustomize Secret Generator Plugins
+authors:
+  - "@sethpollack"
+owning-sig: sig-cli
+reviewers:
+  - "@monopole"
+  - "@Liujingfang1"
+approvers:
+  - "@monopole"
+  - "@Liujingfang1"
+editor: "@sethpollack"
+creation-date: 2019-02-04
+last-updated: 2019-02-04
+status: provisional
+---
+
+# Kustomize Secret Generator Plugins
+
+## Table of Contents
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+
+## Summary
+
+Kustomize removed the `commands` feature from `SecretGenerator` due to security concerns. This proposal is proposing a safe alternative using golang plugins.
+
```

Comment:
link to golang plugins, and maybe a sentence or two as to why they are different from any old executable

Reply:
From the go plugins doc:

> Currently plugins are only supported on Linux and macOS.

Shouldn't the solution work on Windows since [kustomize ships for Windows](https://github.com/kubernetes-sigs/kustomize/releases)?

Reply:
https://github.com/kubernetes-sigs/kustomize/issues/791